File: Python/stringReversal.py
def reverse_string(s):
    """Inefficient string reversal using manual loops"""
    reversed_str = ""
    for i in range(len(s) - 1, -1, -1):  # Iterates character by character
        reversed_str += s[i]  # String concatenation in loop (O(n²) complexity)

    return reversed_str

text = "This is an inefficient string reversal"
print(reverse_string(text))

File: JavaScript/stringReversal.js
function reverseString(str) {
    let reversed = "";
    for (let i = str.length - 1; i >= 0; i--) { // Loops manually
        reversed += str[i]; // String concatenation in loop (slow in JS)
    }
    return reversed;
}

console.log(reverseString("JavaScript"));
